From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Tue, 4 Feb 2025 15:12:22 +0000
Subject: [PATCH] disable unused histogram checks for adding site settings

Added NUM_TYPES_CHROMIUM to indicate the content setting type boundary
to avoid the need to update the histogram map.
---
 .../core/browser/content_settings_uma_util.cc | 19 +++++++++++++++++--
 .../core/common/content_settings_types.mojom  |  7 +++++++
 2 files changed, 24 insertions(+), 2 deletions(-)

diff --git a/components/content_settings/core/browser/content_settings_uma_util.cc b/components/content_settings/core/browser/content_settings_uma_util.cc
index a920b1dd191cb..4c69eb48ed8f1 100644
--- a/components/content_settings/core/browser/content_settings_uma_util.cc
+++ b/components/content_settings/core/browser/content_settings_uma_util.cc
@@ -16,16 +16,26 @@ namespace content_settings_uma_util {
 
 void RecordContentSettingsHistogram(const std::string& name,
                                     ContentSettingsType content_setting) {
+  if (content_setting >= ContentSettingsType::NUM_TYPES_CHROMIUM) {
+    // We don't store histogram values for donwstream added site settings.
+    return;
+  }
+
   base::UmaHistogramExactLinear(
       name, ContentSettingTypeToHistogramValue(content_setting),
       content_settings::kContentSettingsTypeToHistogramValueMax + 1);
 }
 
 int ContentSettingTypeToHistogramValue(ContentSettingsType content_setting) {
+  if (content_setting >= ContentSettingsType::NUM_TYPES_CHROMIUM) {
+    // We don't store histogram values for donwstream added site settings.
+    return -1;
+  }
+
   static_assert(
       content_settings::kContentSettingsTypeToHistogramValue.size() ==
-          // DEFAULT is not in the histogram, so we want [0, kMaxValue]
-          1 + static_cast<size_t>(ContentSettingsType::kMaxValue),
+          // DEFAULT is not in the histogram, so we want [0, NUM_TYPES_CHROMIUM - 1]
+          static_cast<size_t>(ContentSettingsType::NUM_TYPES_CHROMIUM),
       "Update content settings histogram lookup");
 
   auto found = content_settings::kContentSettingsTypeToHistogramValue.find(
@@ -40,6 +50,11 @@ int ContentSettingTypeToHistogramValue(ContentSettingsType content_setting) {
 
 void RecordContentSettingChange(ContentSetting content_setting_value,
                                 ContentSettingsType content_setting_type) {
+  if (content_setting_type >= ContentSettingsType::NUM_TYPES_CHROMIUM) {
+    // We don't store histogram values for donwstream added site settings.
+    return;
+  }
+
   switch (content_setting_value) {
     case CONTENT_SETTING_ALLOW:
       content_settings_uma_util::RecordContentSettingsHistogram(
diff --git a/components/content_settings/core/common/content_settings_types.mojom b/components/content_settings/core/common/content_settings_types.mojom
index e0129fbbea160..ebe846d8370e1 100644
--- a/components/content_settings/core/common/content_settings_types.mojom
+++ b/components/content_settings/core/common/content_settings_types.mojom
@@ -493,5 +493,12 @@ enum ContentSettingsType {
 
   // Stores allowlist decisions for Suspicious Site Warnings.
   SUSPICIOUS_SITE_WARNING_DATA,
+
+  // Indicator for latest upstream content settings type.
+  // Currently, histograms are unused and planned to be stubbed in the near future.
+  NUM_TYPES_CHROMIUM,
+  // Three line placeholder comment to minimize cascading conflicts.
+  // Three line placeholder comment to minimize cascading conflicts.
+  // Three line placeholder comment to minimize cascading conflicts.
 };
 // LINT.ThenChange(//components/content_settings/core/common/content_settings_uma_constants.h:kContentSettingsTypeToHistogramValue)
